src/java/org/apache/cassandra/repair/RepairRunnable.java

```diff
@@ -363,27 +381,60 @@ private NeighborsAndRanges getNeighborsAndRanges()
         return new NeighborsAndRanges(shouldExcludeDeadParticipants, allNeighbors, commonRanges);
     }
 
+    @VisibleForTesting
+    public static NeighborsAndRanges createNeighbordAndRanges(RepairOption options)
+    {
+        if (options.getHosts().size() <= 0)
+            throw new IllegalArgumentException("There should be at least 1 host when validateNeighbors is set to true");
```

Comment:
what is `validateNeighbors`? it appears only in these two assertions.

furthermore, this function is called when we skip neighbour validation, so it sounds more like `false` would be more appropriate?

Reply:
also, why explicit throw instead of `Preconditions.check....` ? any reason for that?